anthos-onprem-terraform/avmw_user_cluster_control_plane_v2/main.tf

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#[START anthos_onprem_terraform_vmware_user_cluster_control_plane_v2_main]
+module "enable_google_apis_primary" {
+  source     = "terraform-google-modules/project-factory/google//modules/project_services"
+  version    = "~> 14.0"
+  project_id = var.project_id
+  activate_apis = [
+    "cloudresourcemanager.googleapis.com",
+    "anthos.googleapis.com",
+    "anthosgke.googleapis.com",
+    "container.googleapis.com",
+    "gkeconnect.googleapis.com",
+    "gkehub.googleapis.com",
+    "serviceusage.googleapis.com",
+    "stackdriver.googleapis.com",
+    "monitoring.googleapis.com",
+    "logging.googleapis.com",
+    "iam.googleapis.com",
+    "compute.googleapis.com",
+    "anthosaudit.googleapis.com",
+    "opsconfigmonitoring.googleapis.com",
+    "file.googleapis.com",
+    "connectgateway.googleapis.com"
+  ]
+  disable_services_on_destroy = false
+}
+
+# Enable GKE OnPrem API
+resource "google_project_service" "default" {
+  project            = var.project_id
+  service            = "gkeonprem.googleapis.com"
+  disable_on_destroy = false
+}
+
+# This module is used to update the platform controller on your admin cluster. This
+# is a necessary step for the user cluster version update. If the admin cluster is 
+# already on the correct version, then this module does not change anything
+module "gcloud_update_admin_cluster_platform_controller" {
+  source                = "terraform-google-modules/gcloud/google"
+  version               = "~> 2.0"
+  platform              = "linux"
+  create_cmd_entrypoint = "gcloud"
+  create_cmd_body       = <<EOT
+    beta container vmware admin-clusters               \
+    update ${var.admin_cluster_name}                   \
+    --required-platform-version=${var.on_prem_version} \
+    --project ${var.project_id}                        \
+    --location ${var.region}
+  EOT
+}
```

Comment:
it looks like this module assume existence of the admin cluster on VMWare. what happens if such cluster does not exists?